escrow/atomic_storage.go

```diff
@@ -33,6 +33,23 @@ type PrefixedAtomicStorage struct {
 	keyPrefix string
 }
 
+//It is recommended to use this function to create a PrefixedAtomicStorage
+func NewPrefixedAtomicStorage(atomicStorage AtomicStorage, prefix string) *PrefixedAtomicStorage {
+	 passedStorage, ok := atomicStorage.(*PrefixedAtomicStorage) ;
+	if ok && passedStorage != nil {
```

Comment:
@vsbogd , dont like this type check here , any other suggestions ?

Reply:
we cant create PrefixedAtomicStorage in components.go unless delegate, is made as Delegate,However we dont need the type casting ;), looked weird to even do on the first place .. have a test case to check the nested paths , i think code looks neater and simpler now , thanks @vsbogd